src/zkbob/ZkBobCompoundingMixin.sol

```diff
@@ -0,0 +1,168 @@
+     */
+    function rebalance(uint256 minRebalanceAmount, uint256 maxRebalanceAmount) external {
+        if (maxRebalanceAmount < minRebalanceAmount) {
+            (minRebalanceAmount, maxRebalanceAmount) = (maxRebalanceAmount, minRebalanceAmount);
+        }
+
+        YieldParams storage params = yieldParams;
+        IERC4626 yieldVault = IERC4626(params.yield);
+
+        if (address(yieldVault) == address(0)) {
+            return;
+        }
+
+        address operator = params.yieldOperator;
+        require(
+            operator == address(0) || operator == msg.sender,
+            "ZkBobCompoundingPool: Rebalance is an operator-called method"
+        );
+
+        uint256 underlyingBalance = IERC20(token).balanceOf(address(this));
+        uint256 buffer = params.buffer;
+
+        uint256 balancesDiff = (underlyingBalance >= buffer) ? underlyingBalance - buffer : buffer - underlyingBalance;
+        balancesDiff = (balancesDiff >= minRebalanceAmount)
+            ? ((balancesDiff <= maxRebalanceAmount) ? balancesDiff : maxRebalanceAmount)
+            : 0;
+        if (balancesDiff <= params.dust) {
+            return;
+        }
+        if (underlyingBalance > buffer) {
+            investedAssetsAmount += balancesDiff;
+            investedSharesAmount += yieldVault.deposit(balancesDiff, address(this));
+            emit Rebalance(0, balancesDiff);
+        } else {
+            investedAssetsAmount -= balancesDiff;
+            investedSharesAmount -= yieldVault.withdraw(balancesDiff, address(this), address(this));
+            emit Rebalance(balancesDiff, 0);
+        }
+    }
+
+    /**
+     * @dev Collects accumulated fees and generated yield for the specific collateral.
+     * Callable only by the contract owner / proxy admin / yield admin.
+     * @param minClaimAmount minimum amount of token to claim.
+     * @return Claimed amount.
+     */
+    function claim(uint256 minClaimAmount) external returns (uint256) {
+        YieldParams storage params = yieldParams;
+
+        IERC4626 yieldVault = IERC4626(params.yield);
+
+        if (address(yieldVault) == address(0)) {
+            return 0;
+        }
+
+        address operator = params.yieldOperator;
+        require(
+            operator == address(0) || operator == msg.sender, "ZkBobCompoundingPool: Claim is an operator-called method"
+        );
+
+        uint256 dust = params.dust;
+        minClaimAmount = minClaimAmount > dust ? minClaimAmount : dust;
+
+        uint256 currentInvestedSharesAmount = investedSharesAmount;
+
+        uint256 toClaimAmount = yieldVault.convertToAssets(currentInvestedSharesAmount) - investedAssetsAmount;
+
+        if (toClaimAmount < minClaimAmount) {
+            return 0;
+        }
+
+        uint256 deltaShares = yieldVault.withdraw(toClaimAmount, address(this), address(this));
+        investedSharesAmount = currentInvestedSharesAmount - deltaShares;
```

Comment:
I am not sure, if there is no rounding problems.
Maybe it is better to pass shares as an argument and add check on balance in the code